Makefile

```diff
@@ -92,7 +92,7 @@ ifneq ($(wildcard $(ASSETS_DIR)/.),)
 	cp -r $(ASSETS_DIR) dist/$(PLUGIN_ID)/
 endif
 ifneq ($(HAS_PUBLIC),)
-	cp -r public/ dist/$(PLUGIN_ID)/
+	cp -r public/ dist/$(PLUGIN_ID)/public/
```

Comment:
Could you elaborate a bit why this is needed? What is the bug this code change fixes?

Reply:
The bug is that files in `public/` don't actually get served by mattermost - they go into the tarball and are ignored by mattermost.

If we put them in the `public/` dir in the packaged plugin tarball, then mattermost starts hosting them under the plugin's URL, which is what we are meant to be demonstrating here.

Reply:
@grubbins On my local machine (Linux) the generated tarball is exactly the same with tar ball. Maybe this is a shell specific issue?

Reply:
https://github.com/mattermost/mattermost-plugin-starter-template/pull/97#discussion_r436647294

Reply:
On my local machine (linux, bash) both your branch and `master` generate exactly the same tarball. Maybe the issue you describe is caused by a shell specific behaviour?

Reply:
Aha, you're right, the issue occurred on OSX, which behaves differently to Linux - see `man cp` on OSX:
```
If the source_file ends in a /, the contents of the directory are copied rather than the directory itself.
```

Another version that I believe would work for both:
```
cp -R public dist/$(PLUGIN_ID)/
```
IMO this is actually clearer too. When I see the trailing slash in the first argument I have to think about the effect (e.g. `rsync` treats that differently even on Linux). With no trailing slash it's very clear that we are copying the dir itself to the destination.

(note `-R` does the same as `-r` on Linux, but on OSX, `-r` is deprecated in favour of `-R` - hence suggesting `-R`)

Reply:
Thanks for investigating. :+1: for omitting the trailing slash.

Could you please check if there are other instances of this issue in `Makefile`?